bots/flakes-refresh

```diff
@@ -0,0 +1,166 @@
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import sys
+import time
+import os
+import urllib
+import socket
+import json
+
+sys.dont_write_bytecode = True
+
+import task
+
+NUMBER_OPEN_ISSUES = 7            # How many issues do we want to have open at a given time?
+
+# How far back does our data go?  If a flake gets fixed but is still
+# flaky after this many days, the bots open another issue.
+
+WINDOW_DAYS = 90
+
+# This parses the output JSONL format discussed here, where various
+# values are grouped:
+#
+# https://github.com/cockpit-project/cockpituous/blob/master/learn/README.md
+
+# Here we're looking for a field in a record that only has one value
+def value(record, field):
+    values = record.get(field, [])
+    if len(values) == 1:
+        return values[0][0] or ""
+    return None
+
+# Here we're looking for the count of a specific field/value in the record
+def count(record, field, only):
+    values = record.get(field, [])
+    for value, count in values:
+        if value != only:
+            continue
+        return count
+    return 0
+
+def run(api, verbose=False, dry=False):
+    open_issues = api.issues(labels=[ "flake" ])
+    create_count = NUMBER_OPEN_ISSUES - len(open_issues)
+
+    if create_count <= 0:
+        return 0
+
+    if verbose:
+        sys.stderr.write("Going to create %s new flake issue(s)\n" % create_count)
+
+    host = os.environ.get("COCKPIT_LEARN_SERVICE_HOST")
+    if not host:
+        raise RuntimeError("no learn host environment variable: $COCKPIT_LEARN_SERVICE_HOST")
+    port = os.environ.get("COCKPIT_LEARN_SERVICE_PORT", "8080")
+    url = "http://{0}:{1}/active/statistics.jsonl".format(host, port)
+
+    statistics = [ ]
+
+    # Retrieve the URL
+    try:
+        req = urllib.request.Request(url)
+        with urllib.request.urlopen(req) as f:
+            for line in f.readlines():
+                try:
+                    record = json.loads(line.decode('utf-8'))
+                    statistics.append(record)
+                except ValueError as ex:
+                    sys.stderr.write("{0}: {1}\n".format(url, ex))
+    except (ConnectionResetError, urllib.error.URLError, socket.gaierror) as ex:
+        sys.stderr.write("{0}: {1}\n".format(url, ex))
+        return False
+
+    tests = { }
+
+    for record in statistics:
+        test = value(record, "test")
+        context = value(record, "context")
+        status = value(record, "status")
+        tracker = value(record, "tracker")
+
+        # There should only be one of all of these values
```

Comment:
Reading this sounds like "make sure that *not*  both exist", while the  code says "make sure that both *do* exist".